src/Server/TCPHandler.cpp

```diff
@@ -2175,6 +2177,22 @@ void TCPHandler::processQuery(std::optional<QueryState> & state)
     /// so we have to apply the changes first.
     state->query_context->setCurrentQueryId(state->query_id);
 
+    /// After query execution, check if we need to close the connection
```

Comment:
It will happen before query execution actually (see that `processQuery` happens in the loop of `runImpl` before `executeQuery`), you should put it at some point after finishing processing the query